src/runtime_src/core/common/api/xrt_hw_context.cpp

```diff
@@ -114,6 +117,12 @@ class hw_context_impl : public std::enable_shared_from_this<hw_context_impl>
   {
     return m_hdl.get();
   }
+
+  std::shared_ptr<xrt_core::usage_metrics::base_logger>
+  get_usage_logger()
+  {
+    return m_usage_logger;
+  }
```

Comment:
I understand the point of keeping the user_logger as a shared_ptr data member, but this function should return `xrt_core::usage_metrics::base_logger*`, because callers will not share ownership.  Please change all such functions in other compilation units also if there are any same functions.
```suggestion
  xrt_core::usage_metrics::base_logger*
  get_usage_logger()
  {
    return m_usage_logger.get();
  }
```